src/skvhttpclient/src/skvhttp/dto/SKVRecord.cpp

```diff
@@ -278,8 +242,12 @@ SKVRecord SKVRecord::getSKVKeyRecord() {
     size_t numKeyFields = schema->partitionKeyFields.size() + schema->rangeKeyFields.size();
     for (size_t i = 0; i < schema->fields.size(); ++i) {
         if (i < numKeyFields) {
-            K2_DTO_CAST_APPLY_FIELD_VALUE(_fieldApplier, schema->fields[i], builder, *this);
+            // the partition and range keys are first in the record
+            visitNextField([&builder] (auto&, auto&& value) {
```

Comment:
Same issue as above, I think this won't work if the key fields are not in order.

Reply:
I'm not sure this is the same. The order is governed by the schema. The fields however all need to be in the front. This just copies them to the new record, in the same position as they are in the source record. Let's chat to make sure

Reply:
You are right, this should work.